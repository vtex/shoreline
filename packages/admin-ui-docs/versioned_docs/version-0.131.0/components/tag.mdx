---
title: Tag
path: /tag/
hide_table_of_contents: true
---

import { Page, Panel } from '@site/src/components/Page'

export const tabToc = {
  overview: toc.filter((item) =>
    ['Examples', 'Sizes', 'Palettes', 'Deletable', 'Icon'].includes(item.value)
  ),
  codeReference: toc.filter((item) => ['Usage', 'Props'].includes(item.value)),
  designReference: toc.filter((item) => ['Doâ€™s'].includes(item.value)),
}

<Page toc={tabToc} hasTabs>
  <Panel id="overview">

Tags represent a status, or a common denominator. They make sections and entities quickly identifiable and searchable.

## Examples

### Sizes

By default, the size property has a `regular` value, but you can also set it to `small`.

```jsx live
<>
  <Tag label="Rio de Janeiro" />
  <Tag label="Rio de Janeiro" size="small" />
</>
```

### Palettes

By default, the palette property has the `lightBlue` value, but you can also set it to the following values: `red`, `orange`, `green`, `cyan`, `teal`, `gray` and `purple`.

```jsx live
<>
  <Tag label="Rio de Janeiro" />
  <Tag label="Rio de Janeiro" palette="red" />
  <Tag label="Rio de Janeiro" palette="orange" />
  <Tag label="Rio de Janeiro" palette="green" />
  <Tag label="Rio de Janeiro" palette="red" />
  <Tag label="Rio de Janeiro" palette="cyan" />
  <Tag label="Rio de Janeiro" palette="teal" />
  <Tag label="Rio de Janeiro" palette="gray" />
  <Tag label="Rio de Janeiro" palette="purple" />
</>
```

### Deletable

You can add the `handleDelete` property to configure the tag as deletable. When this function is defined the tag will have a `button` where you can handle click events.

```jsx live
<Tag label="Rio de Janeiro" handleDelete={() => window.alert('Tag deleted')} />
```

### Icon

You can add one Icon on the left side of the `Tag`. Just use the `icon` property.

```jsx live
<Tag icon={<IconHeart />} label="Rio de Janeiro" />
```

  </Panel>
  <Panel id="codeReference">

## Usage

```jsx isStatic
import { Tag } from '@vtex/admin-ui'

function Example() {
  return (
    <Tag
      icon={<IconHeart />}
      label="Here goes the label!"
      handleDelete={() => window.alert('Tag deleted')}
    />
  )
}
```

## Props

| Name         | Type                                                              | Description                       | Required | Default            |
| ------------ | ----------------------------------------------------------------- | --------------------------------- | -------- | ------------------ |
| label        | `string`                                                          | Tag size                          | âœ…       | -                  |
| csx          | `StyleObject`                                                     | Custom styles                     | ðŸš«       | `{}`               |
| palette      | `'lightBlue', 'green', 'red', 'orange', 'cyan', 'teal', 'purple'` | Tag theme                         | ðŸš«       | `'blue'`, `'gray'` |
| size         | `'regular', 'small'`                                              | Tag size                          | ðŸš«       | `'regular'`        |
| handleDelete | `() => void`                                                      | When defined the tag is deletable | ðŸš«       | -                  |
| icon         | `ReactNode`                                                       | Tag icon                          | ðŸš«       | -                  |

  </Panel>
  <Panel id="designReference">

## Doâ€™s

1. Use semantic colors that have a clear and contextualized meaning in your application, such as green for success and red for errors. However, if your application employs a richer palette with unambiguous usage of colors don't hesitate to use your Tags with them.
2. They should always be read-only.

ðŸ‘‰ For UX Writing guidelines, access https://uxwriting.vtex.com

  </Panel>
</Page>
