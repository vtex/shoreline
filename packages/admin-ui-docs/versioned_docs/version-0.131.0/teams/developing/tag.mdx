---
title: Tag
path: /guidelines/tag/
sidebar_position: 5
---

The `tag` is used to enhance components or jsx elements with admin-ui styles. Before proceeding, make sure that you understand the [csx concept](#).

## Jsx elements

The most common use-case is the jsx elements:

```jsx live
<tag.div
  csx={{
    bg: 'muted',
    color: 'base',
    border: 'default',
    borderRadius: 'default',
    padding: 4,
  }}
>
  admin-ui-powered div
</tag.div>
```

## Function

You can also use tag as a function for elements or components:

```jsx noInline live
const Section = tag('section') // jsx tags are allowed
const Button = tag(ReakitButton) // also components

function Example() {
  return (
    <Section csx={{ padding: 4, ...palette('purple') }}>
      admin-ui-powered section
      <Button
        csx={{
          padding: 2,
          cursor: 'pointer',
          bg: 'action.main.text',
          color: 'action.main.text',
          ':hover': {
            bg: 'action.main.textHover',
            color: 'action.main.textHover',
          },
          ':active': {
            bg: 'action.main.textPressed',
            color: 'action.main.textPressed',
          },
        }}
      >
        Button
      </Button>
    </Section>
  )
}

render(<Example />)
```

## Polymorphism

A `tag` jsx element can be rendered as any HTML tag or component. To use this variation, add a valid component type to the `as` property.

```jsx live
function Example() {
  const menu = useReakitMenuState()

  return (
    <>
      <tag.button
        as={ReakitMenuButton}
        {...menu}
        csx={{
          height: 32,
          borderRadius: 4,
          padding: 2,
          cursor: 'pointer',
          bg: 'action.main.text',
          color: 'action.main.text',
          ':hover': {
            bg: 'action.main.textHover',
            color: 'action.main.textHover',
          },
          ':active': {
            bg: 'action.main.textPressed',
            color: 'action.main.textPressed',
          },
        }}
      >
        Preferences
      </tag.button>
      <tag.div
        as={ReakitMenu}
        {...menu}
        aria-label="Preferences"
        csx={{
          border: '1px solid',
          borderColor: 'popover',
          display: 'flex',
          flexDirection: 'column',
          bg: 'popover',
          borderRadius: 4,
          '> button': {
            borderRadius: 4,
            margin: 1,
            height: 32,
            cursor: 'pointer',
            ...listBoxItem('main'),
            ...focusVisible('main'),
          },
        }}
      >
        <tag.button as={ReakitMenuItem} {...menu}>
          Settings
        </tag.button>
        <tag.button as={ReakitMenuItem} {...menu}>
          Extensions
        </tag.button>
        <tag.button as={ReakitMenuItem} {...menu}>
          Keyboard shortcuts
        </tag.button>
      </tag.div>
    </>
  )
}
```
