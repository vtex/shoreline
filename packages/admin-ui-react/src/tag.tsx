import React from 'react'
import { isFunction, capitalize } from '@vtex/admin-ui-util'

import { useSystem } from './context'
import type { CsxCall, ExtractHTMLAttributes, RenderProp } from './types'
import type { DOMElements } from './domElements'
import { domElements } from './domElements'

/**
 * Creates AdminUI jsx elements
 * @param type jsx tag
 * @example
 * const Box = _jsx('div')
 *
 * // with styles
 * <Box csx={{ bg: 'pink' }} />
 *
 * // polymorphism
 * import { Link } from 'gatsby'
 *
 * <Box as={Link} to="/">Gatsby Link</Box>
 */
function _tag<T extends React.ElementType, Props>(
  type: T
): AdminUITagElement<T, Props> {
  function Element(props: ElementPropsWithAs<Props, T>, ref: React.Ref<T>) {
    const {
      as: ElementCall = type,
      children,
      className,
      csx = {},
      ...forwardProps
    } = props

    const { cn, cx } = useSystem()

    return (
      <ElementCall
        ref={ref}
        className={cx(className, cn(csx))}
        {...(forwardProps as any)}
      >
        {isFunction(children) ? children(forwardProps) : children}
      </ElementCall>
    )
  }

  Element.displayName = capitalize(String(type) ?? 'AdminUI')

  return React.forwardRef(Element) as unknown as AdminUITagElement<T, Props>
}

/**
 * AdminUI jsx elements factory
 * @example
 * // standalone
 * const Box = tag('div')
 *
 * // rendering elements (you can replace div for any valid jsx tag)
 * <tag.div csx={{ color: 'pink' }}>AdminUI styled div</tag.div>
 *
 * // polymorphism
 * <tag.button as={ExternalLibButton} csx={{ bg: 'pink' }} >ðŸ¦„ this is magic</tag.button>
 */
export const tag = _tag as typeof _tag &
  {
    [Tag in DOMElements]: AdminUITagElement<Tag, {}>
  }

domElements.forEach((t) => {
  tag[t] = _tag(t)
})

export type AdminUITagElement<T extends React.ElementType, Props> = {
  <TT extends React.ElementType>(
    props: ElementPropsWithAs<Props, TT> & { as: TT }
  ): JSX.Element
  (props: ElementPropsWithAs<Props, T>): JSX.Element
}

export type ElementPropsWithAs<Props, T extends React.ElementType> = Props &
  Omit<React.ComponentPropsWithRef<T>, 'as' | keyof Props> &
  CsxCall & {
    className?: string
    as?: T
    children?: React.ReactNode | RenderProp<ExtractHTMLAttributes<any>>
  }
