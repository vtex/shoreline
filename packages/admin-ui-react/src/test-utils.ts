import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
