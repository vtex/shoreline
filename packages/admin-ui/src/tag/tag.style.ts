import { palette, style, styleVariants } from '@vtex/admin-ui-core'

export const baseline = style({
  display: 'inline-flex',
  alignItems: 'center',
  borderRadius: '100px',
  text: '$body',
  border: 'none',
  '> svg:nth-of-type(1)': {
    marginRight: 1,
  },
  height: 40,
  paddingX: '4',
  svg: {
    width: 20,
    minWidth: 20,
    height: 20,
    minHeight: 20,
  },
})

export const variants = styleVariants({
  palette: {
    lightBlue: palette('lightBlue'),
    green: palette('green'),
    orange: palette('orange'),
    cyan: palette('cyan'),
    purple: palette('purple'),
    teal: palette('teal'),
    red: palette('red'),
    gray: palette('gray'),
  },
})
