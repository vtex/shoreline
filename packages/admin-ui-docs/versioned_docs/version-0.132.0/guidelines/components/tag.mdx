---
title: Tag
path: /tag/
hide_table_of_contents: true
---

import { Page, Panel } from '@site/src/components/Page'
import { PageSection } from '@site/src/components/PageSection'
import { IntroSection } from '@site/src/components/IntroSection'

export const tabToc = {
  overview: [
    {
      id: 'what-is-this',
      value: 'What is this?',
      level: 2,
    },
    {
      id: 'examples',
      value: 'Examples',
      level: 2,
    },
    {
      id: 'variants',
      value: 'Variants',
      level: 3,
    },
    {
      id: 'sizes',
      value: 'Sizes',
      level: 3,
    },
  ],
  codeReference: [
    {
      id: 'usage',
      value: 'Usage',
      level: 2,
    },
    {
      id: 'props',
      value: 'Props',
      level: 2,
    },
  ],
  designReference: [
    {
      id: 'dos',
      value: "Do's",
      level: 2,
    },
  ],
}

<Page toc={tabToc} hasTabs>
  <Panel id="overview">

<IntroSection>

## What is this?

Tags represent a status, or a common denominator. They make sections and entities quickly identifiable and searchable.

</IntroSection>

<PageSection
  id={tabToc.overview[1].id}
  title={tabToc.overview[1].value}
>

<PageSection
  id={tabToc.overview[2].id}
  title={tabToc.overview[2].value}
  explanation="By default, the variant property has the 'lightBlue' value, but you can also set it to the following values: 'red', 'orange', 'green', 'cyan', 'teal', 'gray', 'purple' and 'pink'."
>

```jsx live
<Inline>
  <Tag label="Short text" />
  <Tag label="Short text" variant="red" />
  <Tag label="Short text" variant="orange" />
  <Tag label="Short text" variant="green" />
  <Tag label="Short text" variant="cyan" />
  <Tag label="Short text" variant="teal" />
  <Tag label="Short text" variant="gray" />
  <Tag label="Short text" variant="purple" />
  <Tag label="Short text" variant="pink" />
</Inline>
```

</PageSection >

<PageSection
  id={tabToc.overview[3].id}
  title={tabToc.overview[3].value}
  explanation="By default, the size property has a 'normal' value, but you can also set it to 'large'."
>

```jsx live
<Inline>
  <Tag label="Short text" size="normal" />
  <Tag label="Short text" size="large" />
</Inline>
```

</PageSection>
</PageSection>
</Panel>

<Panel id="codeReference">

<PageSection
  id={tabToc.codeReference[0].id}
  title={tabToc.codeReference[0].value}
>

```jsx isStatic
import { Tag } from '@vtex/admin-ui'

function Example() {
  return <Tag label="Short text" />
}
```

<PageSection
  id={tabToc.codeReference[1].id}
  title={tabToc.codeReference[1].value}
>

| Name    | Type                                                                      | Description   | Required | Default  |
| ------- | ------------------------------------------------------------------------- | ------------- | -------- | -------- |
| label   | `string`                                                                  | Tag label     | âœ…       | -        |
| csx     | `StyleObject`                                                             | Custom styles | ðŸš«       | `{}`     |
| variant | `lightBlue`, `green`, `red`, `orange`, `cyan`, `teal`, `purple` or `pink` | Tag theme     | ðŸš«       | `gray`   |
| size    | `normal`, `large`                                                         | Tag size      | ðŸš«       | `normal` |

</PageSection >
</PageSection >
</Panel>

<Panel id="designReference">

<PageSection
  id={tabToc.designReference[0].id}
  title={tabToc.designReference[0].value}
>

1. Use semantic colors that have a clear and contextualized meaning in your application, such as green for success and red for errors. However, if your application employs a richer palette with unambiguous usage of colors don't hesitate to use your Tags with them.
2. They should always be read-only.

ðŸ‘‰ For UX Writing guidelines, access https://uxwriting.vtex.com

</PageSection>
</Panel>

</Page>
