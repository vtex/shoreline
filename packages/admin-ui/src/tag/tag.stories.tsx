/* eslint-disable no-console */
import React from 'react'
import type { Meta, Story } from '@storybook/react'

import type { TagProps } from './tag'
import { Tag } from './tag'
import { Stack } from '../stack'
import { Inline } from '../inline'

export default {
  title: 'admin-ui-review/tag',
  component: Tag,
  argTypes: {
    palette: {
      options: ['cyan', 'purple', 'green', 'red', 'orange', 'teal', 'gray'],
      control: { type: 'select' },
    },
  },
} as Meta

export const Playground: Story<TagProps> = (args) => {
  return <Tag {...args} />
}

Playground.args = {
  label: 'Rio de Janeiro',
}

export function Variants() {
  return (
    <Stack>
      <Inline>
        <Tag label="Rio de Janeiro" />
        <Tag label="Rio de Janeiro" palette="cyan" />
        <Tag label="Rio de Janeiro" palette="purple" />
        <Tag label="Rio de Janeiro" palette="green" />
      </Inline>
      <Inline>
        <Tag label="Rio de Janeiro" palette="red" />
        <Tag label="Rio de Janeiro" palette="orange" />
        <Tag label="Rio de Janeiro" palette="teal" />
        <Tag label="Rio de Janeiro" palette="gray" />
      </Inline>
    </Stack>
  )
}
