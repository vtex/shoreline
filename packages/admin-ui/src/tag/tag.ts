import type { ComponentPropsWithRef } from 'react'
import { createComponent, createHook, useElement } from '@vtex/admin-ui-react'
import { baseline, variants } from './tag.style'
import type { VariantProps } from '@vtex/admin-ui-core'

export const useTag = createHook<'div', TagOptions>((props) => {
  const { variant = 'gray', label, ...htmlProps } = props

  return {
    ...htmlProps,
    baseStyle: {
      ...baseline,
      ...variants({
        variant,
      }),
    },
    children: label,
  }
})

/**
 * Tags represent a status, or a common denominator. They make sections and entities quickly identifiable and searchable.
 *
 * @example
 * <Tag
 *   label="Here goes your label!"
 *   variant="red"
 * />
 */
export const Tag = createComponent<'div', TagOptions>((props) => {
  const tagProps = useTag(props)

  return useElement('div', tagProps)
})

export type TagOptions = VariantProps<typeof variants> & {
  /**
   * Tag Label
   */
  label: string
}

export type TagProps = ComponentPropsWithRef<typeof Tag> & TagOptions
