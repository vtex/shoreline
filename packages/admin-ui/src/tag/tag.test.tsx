import React from 'react'
import { render, axe } from '../test-utils'
import { Tag } from './tag'

describe('Tag tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <Tag label="tag-1" data-testid="tag" csx={{ bg: 'coral' }} />
    )

    expect(getByTestId('tag')).toHaveStyleRule('background', 'coral')
  })

  it('should not have a11y violations', async () => {
    const { container } = render(
      <>
        <Tag label="Rio de Janeiro" palette="cyan" />
        <Tag label="Rio de Janeiro" palette="purple" />
        <Tag label="Rio de Janeiro" palette="green" />
        <Tag label="Rio de Janeiro" palette="red" />
        <Tag label="Rio de Janeiro" palette="purple" />
        <Tag label="Rio de Janeiro" palette="orange" />
        <Tag label="Rio de Janeiro" palette="teal" />
        <Tag label="Rio de Janeiro" palette="gray" />
        <Tag label="tag-1" />
      </>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
