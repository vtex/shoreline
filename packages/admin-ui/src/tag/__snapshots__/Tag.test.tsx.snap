// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #D5EFF4;
  color: #093F46;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #EDE8F4;
  color: #462D63;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #DFF1E0;
  color: #1C431E;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #F8E3E3;
  color: #782020;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #EDE8F4;
  color: #462D63;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #FFEBD7;
  color: #5C2C00;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #CDEFE7;
  color: #00463C;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #E6E6E6;
  color: #3B3B3B;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  border: none;
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #DEECF7;
  color: #143C5C;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

<div
    class="emotion-0"
  >
    tag-1
  </div>
</DocumentFragment>
`;
